lib/Service/TmdbAPIService.php

```diff
@@ -234,9 +238,10 @@ public function getImage(string $size, string $imagePath): array {
 	private function getApiKeyV3(?string $userId): string {
 		$adminApiKey = $this->config->getAppValue(Application::APP_ID, 'api_key_v3');
 		if ($userId === null) {
-			return $adminApiKey;
+			return $adminApiKey !== '' ? $this->crypto->decrypt($adminApiKey) : '';
 		}
-		return $this->config->getUserValue($userId, Application::APP_ID, 'api_key_v3', $adminApiKey) ?: $adminApiKey;
+		$userApiKey = $this->config->getUserValue($userId, Application::APP_ID, 'api_key_v3', $adminApiKey) ?: $adminApiKey;
+		return $userApiKey !== '' ? $this->crypto->decrypt($userApiKey): '';
```

Comment:
This is potentially decrypting the admin key twice (if there is no user key).

Reply:
Why? Only the result of condition which is encrypted string yet from config would be decrypted.

Reply:
Nevermind, I've read too fast.